File: C_Sharp_code/Example/WindowsFormsApp/VB_demo08/Form1.vb
Public Class Form1
    Private Sub button3_Click(sender As Object, e As EventArgs) Handles button3.Click
        Dim a As Integer = 2
        a >>= 1
        Console.WriteLine(a)
    End Sub

    Private Sub button4_Click(sender As Object, e As EventArgs) Handles button4.Click
        Dim a As Integer = 2
        a <<= 1
        Console.WriteLine(a)
    End Sub

    Private Sub button5_Click(sender As Object, e As EventArgs) Handles button5.Click
        Dim a As Integer = 2
        a ^= 2 'a的2次方
        Console.WriteLine(a)
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim a As Integer = 1
        Dim b As Integer

        '先判斷左邊是否為true,如果左邊為true才運算右邊 否則左邊為false,右邊不進行運算
        If (a < 1 AndAlso (b = Convert.ToInt32(DateTime.Now) > 2020)) Then
            MessageBox.Show("True")
        Else
            MessageBox.Show("False")
        End If
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Dim a As Integer = 1
        Dim b As Integer

        '先判斷左邊是否為true,如果左邊為true右邊不進行運算 否則左邊為false,才運算右邊
        If (a <= 1 OrElse (b = Convert.ToInt32(DateTime.Now) > 2020)) Then
            MessageBox.Show("True")
        Else
            MessageBox.Show("False")
        End If
    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button6.Click
        Dim t As Integer = 14
        Dim s As String
        s = If(t >= 12, "下午", "上午")
        MessageBox.Show(s)
    End Sub

    Private Sub Button7_Click(sender As Object, e As EventArgs) Handles Button7.Click
        Dim x As Integer
        x = InputBox("輸入一個數字", "視窗")
        Dim s As String = IIf(x Mod 2 = 0, "偶數", "奇數")
        MessageBox.Show(s)
    End Sub
End Class

File: C_Sharp_code/Example/WindowsFormsApp/CS_demo08/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Windows.Forms;


namespace CS_demo08
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            int a = 1;
            int b = 2;
            a = b++;
            Console.WriteLine("{0}{1}", a, b);
            a = ++b;
            Console.WriteLine("{0}{1}", a, b);
        }

        private void button2_Click(object sender, EventArgs e)
        {
            int a = 1;
            int b = 2;
            a = b--;
            Console.WriteLine("{0}{1}", a, b);
            a = --b;
            Console.WriteLine("{0}{1}", a, b);
        }

        private void button3_Click(object sender, EventArgs e)
        {
            //位元移動
            //轉為2進位再向左或向右移動
            int a = 2;
            a >>= 1;//向右移動1個位元
            Console.WriteLine("{0}", a);
        }

        private void button4_Click(object sender, EventArgs e)
        {
            //位元移動
            //轉為2進位再向左或向右移動
            int a = 2;
            a <<= 1;//向左移動1個位元
            Console.WriteLine("{0}", a);
        }

        private void button5_Click(object sender, EventArgs e)
        {
            int a = 1;
            int b = 2;
            int c = a ^ b;//a,b轉為2進位兩者做是否為互斥
            Console.WriteLine("{0}", c);
        }

        private void button6_Click(object sender, EventArgs e)
        {
            int x = Convert.ToInt32(Interaction.InputBox("請輸入一個數字", "視窗"));
            String s;
            s = x % 2 == 0 ? "偶數" : "奇數";
            MessageBox.Show(s);
        }
    }
}
